### ISSUE TITLE
`AccessorProperty` vs `AccessorPropertyDefinition`

### ISSUE THREAD

**Issue by bradzacher** [estree/estree#292]
Just wanted to clarify naming here.
Currently the [`AccessorProperty`](https://github.com/estree/estree/blob/3f4459c9443145d69a8eab21376e928bc9c16c8c/stage3/decorators.md#accessorproperty) node as part of the stage3 decorators proposal doesn't align with other class memebrs like `PropertyDefinition` and `MethodDefinition`.

Did we want to make the name `AccessorPropertyDefinition` for consistency?

**Comment by JLHwung** [comment]
I can provide some background here.

The `AccessorProperty` is largely based on `PropertyDefinition`, they share most the AST structures. I picked `AccessorProperty` over `AccessorPropertyDefinition` because:

- I tried to persuade Babel team into the `AccessorProperty` AST. Apparently it will be harder if named as `AccessorPropertyDefinition`. In the end, the `ClassAccessorProperty` and `ClassAccessorPrivateProperty` are merged into `ClassAccessorProperty`, the AST differences are reduced but not eliminated: only their types are different. I figure it will be easier to support both AST.
- You can say `AccessorProperty` somehow aligns to `StaticBlock`. Again I can't sell `StaticBlockDefinition` to Babel. I could use `ClassStaticBlock` for Babel but then we would have to deal with the AST difference again.

Therefore I lean to the status-quo and personally I prefer the `Definition`-less name for any class elements in the future.

**Comment by bradzacher** [comment]
I think we should really align with consistency with the ESTree spec rather than consistency with whatever babel has done historically.
This seems to be what ESTree really decided when it chose `PropertyDefinition` over `ClassProperty`+`ClassPrivateProperty`. Other examples:
- Not using `Optional*Expression` for optional chaining.
- Not using `PrivateName` for `#private` identifiers.

I agree that it would have been nice to use `StaticBlockDefinition` for consistency. From the looks of it you added `StaticBlock` (https://github.com/estree/estree/pull/231), so I assume in that instance you similarly chose to align with babel's naming for that?

Personally as this is a property, I think would be better that we aligned the name with the existing spec and add `Definition` instead of breaking from the consistency.

**Comment by JLHwung** [comment]
> I agree that it would have been nice to use StaticBlockDefinition for consistency. From the looks of it you added StaticBlock (https://github.com/estree/estree/pull/231), so I assume in that instance you similarly chose to align with babel's naming for that?

StaticBlock is the middle point between `ClassStaticBlock` (Babel's would-be consistent type name) and `StaticBlockDefinition` (ESTree's would-be consistent type name). Another reason is that we don't have such thing for object literal so there isn't `StaticBlockDefinition` vs `StaticBlock`. Same for accessor property IMO.

> Align with consistency with the ESTree spec rather than consistency with whatever babel has done historically.

We can't change the past. But for the future proposal, I will come up with the AST design that different tools would agree upon. The goal is to minimize the AST differences across toolings. ESTree is not a perfect spec, it represents consensus between several tools.

**Comment by ngregory-rbi** [comment]
What is the status of this? It appears `typescript-eslint` is waiting on alignment on this.

**Comment by nzakas** [comment]
ESTree is already irredeemably inconsistent with how nodes are named. And as I've mentioned on many other issues in this repo, ESTree is not a spec, its an agreement between AST producers to do things a certain way. If @JLHwung can't convince the Babel team to adopt something, then it's already dead in the water and we can't move forward. If `AccessorProperty` is what they'll accept, then that's what it will need to be.
